### ISSUE TITLE
Sort DMD modes by corresponding eigenvalues

### ISSUE THREAD

**Issue by fAndreuzzi** [mathLab/PyDMD#163]
It's convenient for several application (for instance when comparing DMD modes obtained from the same parameterized physical system) to sort DMD modes according to some (persistent) metric.

I propose to add a new optional flag to DMD to perform this task automatically.

**Comment by shervinsahba** [comment]
Something I've used in my own scripts is to sort by the magnitude of the mode amplitude's magnitude, seeing as these contribute most to the reconstruction. I have used options to sort by the magnitude of the eigenvalue, but I'm unsure in my own studies as what the use-case for that may be. Not sure how else to sort amongst complex eigenvalues for DMD in a useful way.

Not sure if PyDMD has a sort feature in it, so here's how I added it. I'd call this after fitting DMD on the object.


```    
def sort_by(self, mode="amps"):
        """
        Sorts DMD analysis results for eigenvalues, eigenvectors (modes), and amplitudes_mod (|b|)
        in order of decreasing magnitude, either by "eigs" or "amps".
        """
        if  mode == "eigs":
            indices = np.abs(self.eigs).argsort()[::-1] 
        elif mode == "amps":
            indices = np.abs(self.amplitudes).argsort()[::-1]
        else:
            mode = "default"
            indices = np.arange(len(self.eigs))

        self.eigs = self.eigs[indices]
        self.Atilde = self.Atilde[indices, :]
        self.modes = self.modes[:, indices]
        self.amplitudes = self.amplitudes[indices]
        print("Sorted DMD analysis by {}.".format(mode))
```

**Comment by fAndreuzzi** [comment]
@shervinsahba Thank you very much for your suggestion, I proposed an addition to the class `DMDOperator` which changes the order of eigenvalues/eigenvectors before computing DMD modes, therefore we can save some time since `self.modes` doesn't need to be sorted.

**Comment by fAndreuzzi** [comment]
Fixed in #164.
